### PR TITLE
SPARK-1965 [WEBUI] Spark UI throws NPE on trying to load the app page for non-existent app

### PR DESCRIPTION
**Pull request by srowen**
Don't throw NPE if appId is unknown. @kayousterhout is this a decent enough band-aid for avoiding a full-blown NPE? it should just render empty content instead

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/deploy/master/ui/ApplicationPage.scala [hunk 1]
```diff
@@ -55,6 +60,9 @@ private[spark] class ApplicationPage(parent: MasterWebUI) extends WebUIPage("app
     val app = state.activeApps.find(_.id == appId).getOrElse({
       state.completedApps.find(_.id == appId).getOrElse(null)
     })
+    if (app == null) {
+      return Seq[Node]()
```
Comment (andrewor14) [inline core/src/main/scala/org/apache/spark/deploy/master/ui/ApplicationPage.scala]: should this return some page with an error message?
```
val content = // some HTML
UIUtils.basicSparkPage(content, "Application not found")
```

Comment (srowen) [inline core/src/main/scala/org/apache/spark/deploy/master/ui/ApplicationPage.scala]: Yeah that makes sense. I emulated the message in `HistoryServer`. Really this should result in an HTTP 404, but this simplified API doesn't give access to the `HttpServletResponse`. It wouldn't be a lot of work to pass this around all through the API. Worth it here, you guys think?

**Comment by srowen** [comment]
Jenkins, retest this please

**Comment by kayousterhout** [comment]
I don't think it's worth it to return a 404, and this is more consistent with how we treat stages/jobs that don't exist.  I might say "No running application with ID _____" in stead of application not found, just to be a little more explicit.

**Comment by srowen** [comment]
OK will make that change, and if there are no more objections, will go ahead with this change to patch up this case.

**Pull request closed without merging**